expertise/service/expertise.py

```diff
@@ -141,29 +177,248 @@ def _get_score_and_metadata_dir(self, search_dir, group_scoring=False):
+            token=or_token,
+            baseurl=config.baseurl
+        )
+        openreview_client_v2 = openreview.api.OpenReviewClient(
+            token=or_token,
+            baseurl=config.baseurl_v2
+        )
+        try:
+            execute_create_dataset(openreview_client, openreview_client_v2, config=config.to_json())
+            self.update_status(config, JobStatus.RUN_EXPERTISE)
+            execute_expertise(config=config.to_json())
+            self.update_status(config, JobStatus.COMPLETED)
+        except Exception as e:
+            self.update_status(config, JobStatus.ERROR, str(e))
+            # Re raise exception so that it appears in the queue
+            exception = e.with_traceback(e.__traceback__)
+            raise exception
+
+    def _get_job_name(self, request):
+        job_name_parts = [request.get('name', 'No name provided')]
+        entities = []
+        if request.get('entityA', {}).get('type'):
+            entities.append(request['entityA'])
+        else:
+            job_name_parts.append('No Entity A Type Found')
+        if request.get('entityB', {}).get('type'):
+            entities.append(request['entityB'])
+        else:
+            job_name_parts.append('No Entity B Type Found')
+
+        for entity in entities:
+            if entity['type'] == 'Group':
+                job_name_parts.append(entity.get('memberOf', 'No Group Found'))
+            elif entity['type'] == 'Note':
+                if entity.get('id'):
+                    job_name_parts.append(entity['id'])
+                elif entity.get('invitation'):
+                    job_name_parts.append(entity['invitation'])
+                elif entity.get('withVenueid'):
+                    job_name_parts.append(entity['withVenueid'])
+                else:
+                    job_name_parts.append('No Note Information Found')
+
+        return f'{job_name_parts[0]}: {job_name_parts[1]} - {job_name_parts[2]}'
+
+    def _get_log_from_request(self, request):
+        log = []
+        if request.get('entityA'):
+            log.append(f"Entity A: {json.dumps(request.get('entityA', {}), indent=4)}")
+        if request.get('entityB'):
+            log.append(f"Entity B: {json.dumps(request.get('entityB', {}), indent=4)}")
+
+        return '\n'.join(log)
+
+    def _get_log_from_config(self, config):
+        log = []
+        if config.name:
+            log.append(f"Job name: {config.name}")
+        if config.paper_id:
+            log.append(f"Paper ID: {config.paper_id}")
+        if config.paper_invitation:
+            log.append(f"Paper invitation: {config.paper_invitation}")
+        if config.paper_venueid:
+            log.append(f"Paper venue ID: {config.paper_venueid}")
+        if config.model:
+            log.append(f"Model: {config.model}")
+        if config.model_params:
+            log.append(f"Model params: {json.dumps(config.to_json().get('model_params', {}), indent=4)}")
```

Comment:
One quick thing: we probably also want to log the `match_group` and `alternate_match_group` (if it exists)